internal/pkg/service/common/task/node.go

```diff
@@ -299,20 +301,40 @@ func (n *Node) runTask(logger log.Logger, task Task, cfg Config) (result Result,
 	task.FinishedAt = &finishedAt
 	task.Duration = &duration
 	task.Outputs = result.Outputs
+
+	// Use task outputs in log message and telemetry
+	var attrs []attribute.KeyValue
+	for k, v := range task.Outputs {
+		// Skip nil values
+		if v := reflect.ValueOf(v); !v.IsValid() || (v.Kind() == reflect.Pointer && v.IsZero()) {
+			continue
+		}
+		// Convert value to a string if possible
+		if str, err := cast.ToStringE(v); err == nil {
+			attrs = append(attrs, attribute.String("result_outputs."+k, str))
+		}
+	}
+	sort.SliceStable(attrs, func(i, j int) bool {
+		return attrs[i].Key < attrs[j].Key
+	})
+
```

Comment:
See `trace.go` bellow.

Task can have some `outputs` (Key-Value pairs).

These outputs were correctly added to `span` as attributes (sed `trace.go`), but in the log message, they were serialized in JSON format. I improved it, they are now mapped to attributes also in the log message.